mivideo/mivideo_extract.py

```diff
@@ -67,62 +88,248 @@ def _readTableLastTime(self, tableName: str, tableColumnName: str) -> Union[date
+
+        :return: a dictionary with ValidDataSourceName and last run timestamp
+        """
+
+        self._kalturaInit()
+
+        KALTURA_MAX_MATCHES_ERROR: str = 'QUERY_EXCEEDED_MAX_MATCHES_ALLOWED'
+        tableName: str = 'mivideo_media_created'
+
+        localLogger = logging.getLogger(f'{logger.name}.mediaCreation')
+        localLogger.info('Starting procedure...')
+
+        kClient: KalturaRequestConfiguration = KalturaClient(KalturaConfiguration())
+        kClient.setKs(  # pylint: disable=no-member
+            KalturaSessionService(kClient).start(
+                self.kUserSecret, type=KalturaSessionType.ADMIN, partnerId=self.kPartnerId))
+        kMedia = KalturaMediaService(kClient)
+
+        lastTime: Union[datetime, None] = (
+            self._readTableLastTime(tableName, 'created_at'))
+
+        if (lastTime):
+            localLogger.info(
+                f'Last time found in table: "{lastTime.isoformat()}"')
+        else:
+            lastTime = datetime.fromisoformat(self.defaultLastTimestamp)
+            localLogger.info('Last time not found in table; '
+                             f'using default time: "{lastTime.isoformat()}"')
+
+        createdAtTimestamp: float = lastTime.timestamp()
+
+        kFilter = KalturaMediaEntryFilter()
+        kFilter.createdAtGreaterThanOrEqual = createdAtTimestamp
+        kFilter.categoriesFullNameIn = self.categoriesFullNameIn
+        kFilter.orderBy = KalturaMediaEntryOrderBy.CREATED_AT_ASC
+
+        kPager = KalturaFilterPager()
+        kPager.pageSize = 500  # 500 is maximum
+        kPager.pageIndex = 1
+
+        results: Sequence[KalturaMediaEntry] = None
+        lastCreatedAtTimestamp: Union[float, int] = createdAtTimestamp
+        lastId: Union[str, None] = None
+        numberResults: int = 0
+        queryPageNumber: int = kPager.pageIndex  # for logging purposes
+        totalNumberResults: int = numberResults  # for logging purposes
+        endOfResults = False
+
+        while not endOfResults:
+            try:
+                results = kMedia.list(kFilter, kPager).objects
+            except KalturaException as kException:
+                if (KALTURA_MAX_MATCHES_ERROR in kException.args):
+                    # set new filter timestamp, reset pager to page 1, then continue
+                    kFilter.createdAtGreaterThanOrEqual = lastCreatedAtTimestamp
+                    localLogger.debug(
+                        f'New filter timestamp: ({kFilter.createdAtGreaterThanOrEqual})')
+
+                    # to avoid dupes, also filter out the last ID returned by previous query
+                    # because Kaltura compares createdAt greater than *or equal* to timestamp
+                    kFilter.idNotIn = lastId
+                    kPager.pageIndex = 1
+                    continue
+
+                localLogger.info(f'Other Kaltura API error: "{kException}"')
+                break
```

Comment:
I don't quite understand the `KALTURA_MAX_MATCHES_ERROR` (what does that mean?), but have you not had any issues with other connection-related exceptions? Seems like we might need some re-try behavior?